pkg/compiler/rego/rego.go

```diff
@@ -124,6 +132,79 @@ func (c *CompilerRego) compileSetDefaults(val string, ids ...string) []string {
 	return compiled
 }
 
+func (c *CompilerRego) linearizeContext(stmt *ast.ContextStatement) []*ast.ActionStatement {
+	line := []*ast.ActionStatement{}
+
+	for _, cond := range stmt.Contidions {
+		for _, act := range stmt.Actions {
+			if act.Context == nil {
+				cAction := &ast.ActionStatement{
+					Token:       act.Action.Token,
+					Action:      act.Action,
+					Verb:        stmt.Verb,
+					TypePattern: stmt.TypePattern,
+					Subject:     cond.Subject,
+					WhereClause: cond.Where,
+				}
+
+				if act.Verb != nil {
+					cAction.Verb = act.Verb
+				}
+				if act.Subject != nil {
+					cAction.Subject = act.Subject
+				}
+				if act.TypePattern != nil {
+					cAction.TypePattern = act.TypePattern
+				}
+				if act.Where != nil {
+					if cond.Where == nil {
+						cAction.WhereClause = act.Where
+					} else {
+						cAction.WhereClause = &ast.WhereClause{
+							Token: act.Where.Token,
+							Condition: &ast.InfixCondition{
+								Token:    token.Token{Type: token.AND, Literal: token.AND},
+								Left:     act.Where.Condition,
+								Operator: token.AND,
+								Right:    cond.Where.Condition,
+							},
+						}
+					}
+				}
+
+				line = append(line, cAction)
+			} else {
+				// in case of context in action
+				ctx := act.Context
+				for _, icond := range stmt.Contidions {
+					ctx.Contidions = append(ctx.Contidions, icond)
+				}
+				line = append(line, c.linearizeContext(ctx)...)
+			}
+		}
+	}
+	return line
+}
+
+func (c *CompilerRego) compileContextStatement(stmt *ast.ContextStatement, lineNum int) (string, error) {
```

Comment:
TODO: future pr - lineNum may need sublineNum also, 
statement 1 is broken up into 1a, 1b, 1c, ...